haystack/nodes/prompt/prompt_node.py

```diff
@@ -0,0 +1,822 @@
+                    raise ValueError(
+                        f"torch_dtype should be a torch.dtype or a string with 'torch.' prefix, got {torch_dtype}"
+                    )
+                torch_dtype_resolved = getattr(torch, torch_dtype.strip("torch."))
+            elif isinstance(torch_dtype, torch.dtype):
+                torch_dtype_resolved = torch_dtype
+            else:
+                raise ValueError(f"Invalid torch_dtype value {torch_dtype}")
+            model_input_kwargs["torch_dtype"] = torch_dtype_resolved
+
+        if len(model_input_kwargs) > 0:
+            logger.info("Using model input kwargs %s in %s", model_input_kwargs, self.__class__.__name__)
+
+        self.pipe = pipeline(
+            "text2text-generation",
+            model=model_name_or_path,
+            device=self.devices[0] if "device_map" not in model_input_kwargs else None,
+            use_auth_token=self.use_auth_token,
+            model_kwargs=model_input_kwargs,
+        )
+
+    def invoke(self, *args, **kwargs):
+        """
+        It takes a prompt and returns a list of generated text using the local Hugging Face transformers model
+        :return: A list of generated text.
+        """
+        output = []
+        if kwargs and "prompt" in kwargs:
+            prompt = kwargs.pop("prompt")
+
+            # We might have some uncleaned kwargs, so we need to take only the relevant.
+            # For more details refer to Hugging Face Text2TextGenerationPipeline documentation
+            model_input_kwargs = {
+                key: kwargs[key]
+                for key in ["return_tensors", "return_text", "clean_up_tokenization_spaces", "truncation"]
+                if key in kwargs
+            }
+            output = self.pipe(prompt, max_length=self.max_length, **model_input_kwargs)
+        return [o["generated_text"] for o in output]
+
+    @classmethod
+    def supports(cls, model_name_or_path: str) -> bool:
+        if not all(m in model_name_or_path for m in ["google", "flan", "t5"]):
+            return False
+
+        try:
+            # if it is google flan t5, load it, we'll use it anyway and also check if model loads correctly
+            AutoModelForSeq2SeqLM.from_pretrained(model_name_or_path)
+        except EnvironmentError:
+            return False
+        return True
+
+
+class OpenAIInvocationLayer(PromptModelInvocationLayer):
+    """
+    PromptModelInvocationLayer implementation for OpenAI's GPT-3 InstructGPT models. Invocations are made via REST API.
+    See [OpenAI GPT-3](https://beta.openai.com/docs/models/gpt-3) for more details.
+
+    Note: kwargs other than init parameter names are ignored to enable reflective construction of the class
+    as many variants of PromptModelInvocationLayer are possible and they may have different parameters
+    """
+
+    def __init__(
+        self, api_key: str, model_name_or_path: str = "text-davinci-003", max_length: Optional[int] = 100, **kwargs
```

Comment:
If the max_length value is optional I think the default value should be None. What do you think?

If we need the default value to be set then I would recommend removing the Optional type.